BTCPayServer/Controllers/UIManageController.APIKeys.cs

```diff
@@ -244,7 +289,50 @@ public async Task<IActionResult> AddApiKey(AddApiKeyViewModel viewModel)
                 Severity = StatusMessageModel.StatusSeverity.Success,
                 Html = StringLocalizer["API key generated!"].Value + $" <code class='alert-link'>{key.Id}</code>"
             });
-            return RedirectToAction("APIKeys");
+            return RedirectToAction(nameof(APIKeys));
+        }
+
+        [HttpGet("~/api-keys/{id}/edit")]
+        public async Task<IActionResult> EditAPIKey(string id)
+        {
+            var key = await _apiKeyRepository.GetKey(id);
+            if (key == null || key.UserId != _userManager.GetUserId(User))
+            {
+                return NotFound();
+            }
+            var viewModel = new EditApiKeyViewModel { Label = key.Label, Id = key.Id };
+            await SetViewModelValues(viewModel);
+            var existingPermissions = key.GetBlob().Permissions;
```

Comment:
To avoid string parsing below, use `Permission.ToPermissions`
```suggestion
            var existingPermissions = Permission.ToPermissions(key.GetBlob().Permissions);
```